edx_exams/apps/lti/views.py

```diff
@@ -12,19 +15,96 @@
+from lti_consumer.lti_1p3.extensions.rest_framework.authentication import Lti1p3ApiAuthentication
+from lti_consumer.lti_1p3.extensions.rest_framework.permissions import LtiProctoringAcsPermissions
 from lti_consumer.models import LtiConfiguration
 from rest_framework import status
 from rest_framework.decorators import api_view, authentication_classes, permission_classes
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 
-from edx_exams.apps.core.api import get_attempt_by_id, update_attempt_status
+from edx_exams.apps.core.api import (
+    get_attempt_by_attempt_number_and_resource_id,
+    get_attempt_by_id,
+    update_attempt_status
+)
 from edx_exams.apps.core.exceptions import ExamIllegalStatusTransition
 from edx_exams.apps.core.statuses import ExamAttemptStatus
 from edx_exams.apps.lti.utils import get_lti_root
 
+log = logging.getLogger(__name__)
+
 EDX_OAUTH_BACKEND = 'auth_backends.backends.EdXOAuth2'
 
+LTI_PROCTORING_ASSESSMENT_CONTROL_ACTIONS = [
+    'pauseRequest',
+    'resumeRequest',
+    'terminateRequest',
+    'update',
+    'flagRequest',
+]
+
+
+@api_view(['POST'])
+@require_http_methods(['POST'])
+@authentication_classes((Lti1p3ApiAuthentication,))
+@permission_classes((LtiProctoringAcsPermissions,))
+def acs(request):
+    """
+    Endpoint for ACS actions
+
+    NOTE: for now just have the actions LOG what the actions is doing.
+    We can implement proper functionality and tests later once we
+    hear back from our third party proctoring service vendors
+    (i.e. Verificient and Proctortrack).
+
+    Currently, we only support flagging of exam attempts.
+    Other ACS actions (pause, resume, terminate, update) could be implemented
+    in the future if desired.
+    """
+    data = json.loads(request.body)
+
+    # This identifies the proctoring tool the request is coming from.
+    user = data['user']
+
+    # The link to exam the user is attempting
+    resource_id = data['resource_link']['id']
+
+    # Exam attempt number (Note: ACS does not differentiate between launches)
+    # i.e, if a launch fails and multiple subsequent launches occur for
+    # the same resource_link + attempt_number combo, the ACS only perceives
+    # this as one singular attempt.
+    attempt_number = data['attempt_number']
+
+    # ACS action to be performed
+    action = data['action']
+
+    # Data validation: Make sure that the exam attempt is either ongoing or completed.
+    # But not verified, rejected, or expired (no need to flag in these cases)
+    # Therefore, any attempt with status from 'started' to 'submitted' can be flagged
+    VALID_STATUSES = [
+        ExamAttemptStatus.started,
+        ExamAttemptStatus.ready_to_submit,
+        ExamAttemptStatus.timed_out,
+        ExamAttemptStatus.submitted,
+    ]
+
+    attempt = get_attempt_by_attempt_number_and_resource_id(attempt_number, resource_id)
```

Comment:
this would also need to look at the user id because attempt_number is not unique. It's always 1 for a user's first attempt, 2 for the second, and so on.

Reply:
What's our error condition if we can't find an attempt for this user and this returns None?